source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm

```diff
@@ -455,6 +456,11 @@ - (void)testSharedEnumsCompatabilityTest
     XCTAssertTrue(static_cast<int>(AdaptiveCards::CardElementType::ToggleInput) == ACRToggleInput);
     XCTAssertTrue(static_cast<int>(AdaptiveCards::CardElementType::Unknown) == ACRUnknown);
     XCTAssertTrue(static_cast<int>(AdaptiveCards::CardElementType::RichTextBlock) == ACRRichTextBlock);
+    XCTAssertTrue(static_cast<int>(AdaptiveCards::ActionType::ShowCard) == ACRShowCard);
+    XCTAssertTrue(static_cast<int>(AdaptiveCards::ActionType::Submit) == ACRSubmit);
+    XCTAssertTrue(static_cast<int>(AdaptiveCards::ActionType::OpenUrl) == ACROpenUrl);
+    XCTAssertTrue(static_cast<int>(AdaptiveCards::ActionType::ToggleVisibility) == ACRToggleVisibility);
+    XCTAssertTrue(static_cast<int>(AdaptiveCards::ActionType::UnknownAction) == ACRUnknownAction);
```

Comment:
what are these tests testing? Any way to annotate the assert statements with descriptions of what the assertion failure signifies?

Reply:
it's making sure that  shared model enum value  matches with ios enum value.

---
In reply to: [336622894](https://github.com/microsoft/AdaptiveCards/pull/3526#discussion_r336622894) [](ancestors = 336622894)

Reply:
aah i see -- missed the point of the test here since I didnt realize it was teh sharedModel enum being compared. Maybe make the test name slightly more obvious? Also 2nd part of my question is to see what happens if the assert fails and the test fails. whats the debugging experience ? Hopefuly we can add a message here that gets spit out into the test logs should teh test fail? 

---
In reply to: [336628762](https://github.com/microsoft/AdaptiveCards/pull/3526#discussion_r336628762) [](ancestors = 336628762,336622894)

Reply:
I will update the test name. once updated, when an assertion fails, i think this should give enough information about why it failed.

---
In reply to: [336631166](https://github.com/microsoft/AdaptiveCards/pull/3526#discussion_r336631166) [](ancestors = 336631166,336628762,336622894)